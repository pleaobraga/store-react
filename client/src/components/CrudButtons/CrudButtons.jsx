import React from 'react'
import PropTypes from 'prop-types'
import './CrudButtons.scss'

export const CrudButtons = ({ create, edit, del }) => {
  return (
    <div className="crud-buttons">
      {create && (
        <button
          key="create"
          className={`create ${create.className ? create.className : ''} `}
          onClick={create.onClick}
        >
          <i className="material-icons create__icon">add_circle</i>
          <span>{create.title ? create.title : 'Add'}</span>
        </button>
      )}
      {edit && (
        <button
          key="edit"
          className={`edit ${edit.className ? edit.className : ''} `}
        >
          <i className="material-icons edit__icon">create</i>
          <span>{edit.title ? edit.title : 'Edit'}</span>
        </button>
      )}
      {del && (
        <button
          key="delete"
          className={`delete ${del.className ? del.className : ''} `}
        >
          <i className="material-icons delete__icon">delete</i>
          <span>{del.title ? del.title : 'Delete'}</span>
        </button>
      )}
    </div>
  )
}

CrudButtons.propTypes = {
  create: PropTypes.shape({
    className: PropTypes.string,
    title: PropTypes.string,
    onClick: PropTypes.func
  }),
  edit: PropTypes.shape({
    className: PropTypes.string,
    title: PropTypes.string,
    onClick: PropTypes.func
  }),
  del: PropTypes.shape({
    className: PropTypes.string,
    title: PropTypes.string,
    onClick: PropTypes.func
  })
}

export default React.memo(CrudButtons)
