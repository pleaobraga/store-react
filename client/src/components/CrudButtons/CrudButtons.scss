@import '../../styles/colors';

.crud-buttons {
  display: flex;
  justify-content: center;
  padding: 2.5rem;

  button {
    display: flex;
    align-items: center;

    cursor: pointer;

    color: white;
    font-weight: 700;

    min-width: 8rem;
    padding: 0.5rem;

    margin-right: 2rem;

    border: 1px solid black;
    border-radius: 2px;

    box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),
      0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);

    span {
      width: 100%;
      text-align: center;
    }

    &.create {
      background-color: $green;
      border: 1px solid $green;
    }

    &.edit {
      background-color: $blue;
      border: 1px solid $blue;
    }

    &.delete {
      background-color: $red;
      border: 1px solid $red;

      width: 7rem;
      height: 4rem;

      &.loading {
        display: block;
      }

      .loader {
        width: 100%;
        height: 100%;
      }
    }

    &:focus {
      outline: none;
    }

    &:active {
      box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2),
        0px 8px 10px 1px rgba(0, 0, 0, 0.14),
        0px 3px 14px 2px rgba(0, 0, 0, 0.12);
    }

    &:disabled,
    &[disabled] {
      border: 1px solid #999999;
      background-color: #cccccc;
      color: #666666;

      &:active {
        box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),
          0px 2px 2px 0px rgba(0, 0, 0, 0.14),
          0px 1px 5px 0px rgba(0, 0, 0, 0.12);
      }
    }
  }
}
