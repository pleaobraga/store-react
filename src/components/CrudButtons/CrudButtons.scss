@import '../../styles/colors';

.crud-buttons {
  display: flex;

  button {
    display: flex;
    align-items: center;

    min-width: 8rem;
    padding: 0.5rem;

    margin-right: 2rem;

    border: 1px solid black;
    border-radius: 2px;

    cursor: pointer;

    color: white;
    font-weight: 700;

    span {
      width: 100%;
      text-align: center;
    }

    &.create {
      background-color: #00796b;
      border: 1px solid #00796b;
    }

    &.edit {
      background-color: #0288d1;
      border: 1px solid #0288d1;
    }

    &.delete {
      background-color: #e64a19;
      border: 1px solid #e64a19;
    }

    &:focus {
      outline: none;
    }
  }
}
