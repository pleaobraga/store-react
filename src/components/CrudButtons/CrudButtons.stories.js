import React from 'react'
import { storiesOf } from '@storybook/react'
import { CrudButtons } from './CrudButtons'

export default storiesOf('Components | CrudButtons', module).add(
  'default',
  () => (
    <CrudButtons
      create={{ onClick: () => {} }}
      edit={{ onClick: () => {} }}
      del={{ onClick: () => {} }}
    />
  ),
  {
    info: {
      inline: true,
      header: false,
      propTables: [CrudButtons]
    }
  }
)
