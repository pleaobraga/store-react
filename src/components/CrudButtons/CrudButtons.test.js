import React from 'react'
import { shallow } from 'enzyme'
import CrudButtons from './CrudButtons'

describe('CrudButtons', () => {
  const crudButtons = shallow(
    <CrudButtons
      create={{ onClick: () => {} }}
      edit={{ onClick: () => {} }}
      del={{ onClick: () => {} }}
    />
  )

  it('render proprely', () => {
    expect(crudButtons).toMatchSnapshot()
  })
})
