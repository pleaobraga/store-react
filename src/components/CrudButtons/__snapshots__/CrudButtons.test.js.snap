// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrudButtons render proprely 1`] = `
<div
  className="crud-buttons"
>
  <button
    className="create  "
    key="create"
    onClick={[Function]}
  >
    <i
      className="material-icons create__icon"
    >
      add_circle
    </i>
    <span>
      Add
    </span>
  </button>
  <button
    className="edit  "
    key="edit"
  >
    <i
      className="material-icons edit__icon"
    >
      create
    </i>
    <span>
      Edit
    </span>
  </button>
  <button
    className="delete  "
    key="delete"
  >
    <i
      className="material-icons delete__icon"
    >
      delete
    </i>
    <span>
      Delete
    </span>
  </button>
</div>
`;
